services/blockchain-indexer/shared/dataService/business/pos/stakers.js

```diff
@@ -53,16 +55,16 @@ const getStakers = async params => {
 		params = remParams;
 
 		params.validatorAddress = address;
-		stakers.meta.validator.address = address;
+		stakersResponse.meta.validator.address = address;
 	}
 
 	if (params.publicKey) {
 		const { publicKey, ...remParams } = params;
 		params = remParams;
 
 		params.validatorAddress = getLisk32AddressFromPublicKey(publicKey);
-		stakers.meta.validator.address = params.validatorAddress;
-		stakers.meta.validator.publicKey = publicKey;
+		stakersResponse.meta.validator.address = params.validatorAddress;
+		stakersResponse.meta.validator.publicKey = publicKey;
```

Comment:
Please ensure that if `params.validatorAddress` is already present at this stage and the derived address from `getLisk32AddressFromPublicKey(publicKey)` is not an exact match, return empty response.

Kindly ensure to add a test case covering the scenario.

Reply:
Applied.